### PR TITLE
Populate self for command input and fix optional secondary files

### PR DESCRIPTION
**Pull request by Horneth**

### CODE REVIEW

**Comment by mcovarr** [comment]
No conformance tests enabled? I'm in a similar predicament over on my Wash U branch where I want to refactor a bunch of stuff but there are no tests to keep me honest. 😢

**Comment by Horneth** [comment]
Yeah those seem like things I would have expected to be failing some conformance tests but apparently not... :/

File: cwl/src/main/scala/cwl/InputParameter.scala [hunk 1]
```diff
@@ -122,6 +123,19 @@ object InputParameter {
...
+            obj.values.toList.traverse[ErrorOr, (String, WomValue)]({
+              case (key, value) => populateFiles(value).map(key -> _)
+            })
+              .map(_.toMap)
+              // transform to Either so we can flatMap
+              .toEither
+              // Validate new types are still valid w.r.t the object
+              .flatMap(WomObject.withTypeChecked(_, obj.womObjectTypeLike))
+              // re-transform to ErrorOr
+              .toValidated
```
Comment (kshakir) [inline cwl/src/main/scala/cwl/InputParameter.scala]: I don't think the method `withTypeChecked` is required.

```scala
// transform to Either so we can flatMap
.toEither
// Validate new types are still valid w.r.t the object
.flatMap(WomObject.withTypeChecked(_, obj.womObjectTypeLike))
// re-transform to ErrorOr
.toValidated
```

should be the same as

```scala
.flatMap(WomObject.withTypeErrorOr(_, obj.womObjectTypeLike))
```

**Comment by cjllanwarne** [comment]
@mcovarr could we make a simple centaur test? Or even a PR into CWL’s suite?

**Comment by cjllanwarne** [comment]
Because otherwise it’s going to be so easy to have accidental regressions on these more surprising edge cases in the future

**Comment by mcovarr** [comment]
@cjllanwarne For my Wash U stuff I've added now added some tests. I'm not sure what's involved for a test on this.

**Comment by Horneth** [comment]
I can add a test.
I'm all for making PRs to CWL conformance tests but it's going to increase the merge time of our PRs if we want to wait for it to be in the CWL repo. Also we'd need to unpin the hash for conformance test or update it every time..

File: cwl/src/main/scala/cwl/InputParameter.scala (hunk 1 above)
Comment (Horneth) [reply cwl/src/main/scala/cwl/InputParameter.scala]: flatMapping `ErrorOr` requires the special `ShortCircuitingFlatMap` import whereas `Either` can naturally flatMap since 2.12 so I prefer using that even though it's a bit more verbose

**Comment by cjllanwarne** [comment]
Thanks @Horneth!

File: src/bin/travis/resources/papi_conformance_expected_failures.txt [hunk 2]
```diff
@@ -114,6 +114,6 @@
 114
 115
 116
-#117
+117
```
Comment (cjllanwarne) [inline src/bin/travis/resources/papi_conformance_expected_failures.txt]: 😢

Comment (Horneth) [reply src/bin/travis/resources/papi_conformance_expected_failures.txt]: It's not as bad as you think, it's a pre-processing conformance test problem where we don't prefix default values in the CWL with gcs paths.
We'll fix it at some point it's just not in the set of #1 priorities now 😄

**Comment by Horneth** [comment]
Looks like I need a formal approval from `one_reviewer` for this one. I assume it means @cjllanwarne or @jbuabbud ?
